pkg/types/validation/installconfig.go

```diff
@@ -529,8 +534,10 @@ func validatePlatform(platform *types.Platform, fldPath *field.Path, network *ty
 func validateProxy(p *types.Proxy, c *types.InstallConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if p.HTTPProxy == "" && p.HTTPSProxy == "" {
-		allErrs = append(allErrs, field.Required(fldPath, "must include httpProxy or httpsProxy"))
+	if c.AdditionalTrustBundlePolicy != types.PolicyAlways {
+		if p.HTTPProxy == "" && p.HTTPSProxy == "" {
+			allErrs = append(allErrs, field.Required(fldPath, "must include httpProxy or httpsProxy"))
+		}
```

Comment:
I am confused by this check. Why does the additionalTrustBundlePolicy affect this requirement? What case is this protecting? To me it seems like we don't need an update here, but perhaps I'm missing something.

Reply:
In case of PolicyAlways, proxy configuration is not a mandatory field. For mitm proxy usecase, there might be no proxy details provided with additionalTrustBundle.

Reply:
This code is only executed when proxy is supplied:
https://github.com/openshift/installer/blob/4d3796fa29b7aa7767376718b318cec893ace4ba/pkg/types/validation/installconfig.go#L122-L124

This function just makes sure that when a proxy is supplied it is valid, the trust bundle policy shouldn't affect that. We shouldn't need any updates to this function. It seems like the current code would break the existing validation in the case of ProxyOnly

Reply:
User can provide proxy details with PolicyAlways configuration. As the policies only control the propagation, proxy details with PolicyAlways configuration will still be valid.

Reply:
Oops, my previous comment had a typo. it should have said "It seems like the current code would break the existing validation in the case of PolicyAlways". 

> User can provide proxy details with PolicyAlways configuration. As the policies only control the propagation, proxy details with PolicyAlways configuration will still be valid.

Right, but your code is saying we have different proxy validation rules based on the policy, which isn't true. Specifically your code says:  when `PolicyAlways`, it is ok to have `noProxy` without `httpProxy` or `httpsProxy`, but when `ProxyOnly` one of `httpProxy` or `httpsProxy` is required (when `noProxy` is set). 

If you removed your change, the logic would be correct: if `noProxy` is set, one of `httpProxy` or `httpsProxy` is required, regardless of policy.

Reply:
Yes, you are right. The condition check is redundant and adds to confusing dependence on policy. Thanks :+1: